src/FakePAB/UtxoParser.hs

```diff
@@ -35,20 +37,23 @@ import Plutus.V1.Ledger.Api (
   CurrencySymbol (..),
  )
 import PlutusTx.Builtins (toBuiltin)
-import Prelude hiding (takeWhile)
+import Prelude hiding (take, takeWhile)
 
 utxoMapParser :: Address -> Parser (TxOutRef, ChainIndexTxOut)
 utxoMapParser address =
   (,) <$> txOutRefParser <* skipSpace <*> chainIndexTxOutParser address
 
 txOutRefParser :: Parser TxOutRef
 txOutRefParser = do
-  txId <- TxId <$> decodeHash (takeWhile (/= ' '))
+  txId <- txIdParser
 
   skipSpace
   txIx <- decimal
   pure $ TxOutRef txId txIx
 
+txIdParser :: Parser TxId
+txIdParser = TxId <$> decodeHash (takeWhile (\c -> c /= ' ' && c /= '\n'))
```

Comment:
If this is ever meant to run on Windows, is should handle `\r` as well. Better use `not . isSpace`.

Reply:
Good point, will do